packages/components/src/internal/components/chart/ChartBuilderModal.tsx

```diff
@@ -0,0 +1,648 @@
+                                                {savedChartModel?.visualizationConfig.chartConfig.measures['y']
+                                                    ?.aggregate?.name ?? 'Sum'}
+                                            </div>
+                                        )}
+                                    </Fragment>
+                                ))}
+                            </div>
+                        </div>
+                    </div>
+                    <div className="row margin-top">
+                        <div className="col-xs-12">
+                            <label>Preview</label>
+                            {previewMsg && <span className="chart-preview-msg gray-text pull-right">{previewMsg}</span>}
+                            {!hasRequiredValues && (
+                                <div className="gray-text">Select required fields to preview the chart.</div>
+                            )}
+                            {hasRequiredValues && (
+                                <div className="chart-preview-body">
+                                    {loadingData && (
+                                        <div className="chart-loading-mask">
+                                            <div className="chart-loading-mask__background" />
+                                            <LoadingSpinner
+                                                wrapperClassName="loading-spinner"
+                                                msg="Loading Preview..."
+                                            />
+                                        </div>
+                                    )}
+                                    <div className="svg-chart__chart" id={divId} ref={ref} />
+                                </div>
+                            )}
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </Modal>
+    );
+});
+
+export const getChartRenderMsg = (chartConfig: ChartConfig, rowCount: number, isPreview: boolean): string => {
+    let msg = '';
+    let sep = '';
+    if (isPreview && rowCount === MAX_ROWS_PREVIEW) {
+        msg = 'The preview is being limited to ' + MAX_ROWS_PREVIEW.toLocaleString() + ' rows.';
+        sep = ' ';
+    }
+
+    if (chartConfig.renderType === 'line_plot' && rowCount > chartConfig.geomOptions.binThreshold) {
+        msg +=
+            sep +
+            ('The number of individual points exceeds ' +
+                MAX_POINT_DISPLAY.toLocaleString() +
+                '. Data points will not be shown on this line plot.');
+    } else if (chartConfig.renderType === 'scatter_plot' && rowCount > MAX_POINT_DISPLAY) {
+        msg +=
+            sep +
+            ('The number of individual points exceeds ' +
+                MAX_POINT_DISPLAY.toLocaleString() +
+                '. The data is now grouped by density.');
+    }
+
+    return msg === '' ? undefined : msg;
+};
```

Comment:
```suggestion
export const getChartRenderMsg = (chartConfig: ChartConfig, rowCount: number, isPreview: boolean): string => {
    const msg = [];
    if (isPreview && rowCount === MAX_ROWS_PREVIEW) {
        msg.push(`The preview is being limited to ${MAX_ROWS_PREVIEW.toLocaleString()} rows.`);
    }

    if (chartConfig.renderType === 'line_plot' && rowCount > chartConfig.geomOptions.binThreshold) {
        msg.push(`The number of individual points exceeds ${MAX_POINT_DISPLAY.toLocaleString()}.`);
        msg.push('Data points will not be shown on this line plot.');
    } else if (chartConfig.renderType === 'scatter_plot' && rowCount > MAX_POINT_DISPLAY) {
        msg.push(`The number of individual points exceeds ${MAX_POINT_DISPLAY.toLocaleString()}.`);
        msg.push('The data is now grouped by density.');
    }

    return msg.length === 0 ? undefined : msg.join(' ');
};
```

If we use `.join` on an array we don't have to track the separator, and if we append sentences to the array we can keep the line length shorter per message which is easier to read IMO.

Reply:
yup, much cleaner. done